src/test/ui/closures/2229_closure_analysis/diagnostics/closure-origin-array-diagnostics.stderr

```diff
@@ -7,7 +7,7 @@ LL |         let [_, _s] = s;
    |                       - closure is `FnOnce` because it moves the variable `s` out of its environment
 LL |     };
 LL |     expect_fn(c);
-   |     --------- the requirement to implement `Fn` derives from here
+   |               - the requirement to implement `Fn` derives from here
```

Comment:
This is slightly misleading. Ideally we would also point at the bound in `expect_fn`. I'm guessing we'd have the Obligation chain in this error, so we could do it.

Reply:
For some reason for closure mismatches, we were skipping noting the obligation cause... let me fix that.